.github/workflows/release_github.yml

```
@@ -43,83 +59,138 @@ jobs:
         run: python3 -m pip install --upgrade ".[publishing]"
       - name: Build eitprocessing
         run: python3 -m build
-  
+    
 
-  github_release:
+  merge_pr:
+    name: Merge the PR into main
+    if: ${{ github.ref_name != 'develop' }}
     needs: test_python_releases
-    runs-on: "ubuntu-latest"
-    defaults:
-      run:
-        shell: bash -l {0}
-
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
     steps:
-      - name: Fail if main branch was selected
-        if: ${{ github.ref_name == 'main' }}
-        run: |
-          echo "Cannot release from main branch, please select valid release branch."
-          exit 1
+      - name: Checkout the code
+        uses: actions/checkout@v4
 
+      - name: Set PR base to main and merge
+        run: |
+          gh pr edit ${{ github.ref_name }} --base main
+          gh pr merge ${{ github.ref_name }} --merge --admin
+    
+  merge_develop:
+    name: Merge develop into main
+    if: ${{ github.ref_name == 'develop'}}
+    needs: test_python_releases
+    runs-on: ubuntu-latest
+    steps: 
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          token: ${{ secrets.GH_PAT }}
           ref: main
           fetch-depth: 0
 
       - name: Configure git
         run: |
-          git config user.email "actions@github.com"
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
```

any reason it needs to be a global setting (idem for `bump-version` job)? The email setting below is not set to global.

```suggestion
          git config user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
```

---

Good points. I think I copied this from two different sources. Generally settings are set globally, but that serves no purpose here.